lib/src/index_impl.dart

```diff
@@ -126,30 +127,32 @@ class MeiliSearchIndexImpl implements MeiliSearchIndex {
     String? highlightPostTag,
     MatchingStrategy? matchingStrategy,
   }) async {
-    final data = <String, Object?>{
-      'q': query,
-      'offset': offset,
-      'limit': limit,
-      'page': page,
-      'hitsPerPage': hitsPerPage,
-      'filter': filter ?? filterExpression?.transform(),
-      'sort': sort,
-      'facets': facets,
-      'attributesToRetrieve': attributesToRetrieve,
-      'attributesToCrop': attributesToCrop,
-      'cropLength': cropLength,
-      'attributesToHighlight': attributesToHighlight,
-      'showMatchesPosition': showMatchesPosition,
-      'cropMarker': cropMarker,
-      'highlightPreTag': highlightPreTag,
-      'highlightPostTag': highlightPostTag,
-      'matchingStrategy': matchingStrategy?.name,
-    };
-    data.removeWhere((k, v) => v == null);
-    final response = await http
-        .postMethod<Map<String, Object?>>('/indexes/$uid/search', data: data);
-
-    return Searcheable.createSearchResult(response.data!);
+    final data = SearchQuery(
+      indexUid: uid,
```

Comment:
Since all the key names repeat, do you think we can make this less verbose?

Reply:
Do you think this change will also improve the user's experience?

Reply:
definitely yes, since having a unified object for search parameters will simplify it for users who want to use multi-search and single index search at the same time

Reply:
not to mention, it's also how it's done in other sdks, e.g. dotnet

Reply:
Let's keep it like this, and in the future, we can change if we felt necessary :)